service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -55,10 +59,37 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             run2740(caseDetails);
         }
 
+        if ("FPLA-2774".equals(migrationId)) {
+            run2774(caseDetails);
+        }
+
         caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
 
+    private void run2774(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
+
+        if ("NE21C50007".equals(caseData.getFamilyManCaseNumber())) {
+            List<Element<HearingBooking>> hearings = caseData.getHearingDetails();
+
+            if (ObjectUtils.isEmpty(hearings)) {
+                throw new IllegalArgumentException("No hearings in the case");
+            }
+
+            if (hearings.size() < 2) {
+                throw new IllegalArgumentException(String.format("Expected 2 hearings in the case but found %s",
+                    hearings.size()));
+            }
+
+            Element<HearingBooking> hearingToBeRemoved = hearings.get(1);
```

Comment:
maybe worth check date of hearing, if hearing(0) will be adjourned or vacated meanwhile then hearing(1) becomes hearing(0) and hearing(2) becomes hearing(1), and wrong one can be removed